### PR TITLE
[FLINK-27283] Add end to end tests for table store

### PR DESCRIPTION
**Pull request by tsreaper**
End to end tests ensure that users can run table store as expected.

### CODE REVIEW

File: flink-table-store-e2e-tests/src/test/resources/log4j2-test.properties [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+# Set root logger level to OFF to not flood build logs
+# set manually to INFO for debugging purposes
+rootLogger.level = INFO
```
Comment (JingsongLi) [inline flink-table-store-e2e-tests/src/test/resources/log4j2-test.properties:21]: `This step has been truncated due to its large size. Download the full logs from the  menu once the workflow run has completed.`
https://github.com/apache/flink-table-store/runs/6070932420?check_suite_focus=true

Comment (JingsongLi) [inline flink-table-store-e2e-tests/src/test/resources/log4j2-test.properties:21]: `This step has been truncated due to its large size. Download the full logs from the  menu once the workflow run has completed.`
https://github.com/apache/flink-table-store/runs/6070932420?check_suite_focus=true

File: flink-table-store-e2e-tests/pom.xml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <artifactId>flink-table-store-parent</artifactId>
+        <groupId>org.apache.flink</groupId>
+        <version>0.2-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>flink-table-store-e2e-tests</artifactId>
```
Comment (JingsongLi) [inline flink-table-store-e2e-tests/pom.xml:31]: Can we have a separate workflow to run e2e?
The overall test run is currently taking too long.

Comment (JingsongLi) [inline flink-table-store-e2e-tests/pom.xml:31]: Can we have a separate workflow to run e2e?
The overall test run is currently taking too long.

File: flink-table-store-e2e-tests/src/test/java/org/apache/flink/table/store/tests/E2eTestBase.java [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+ */
+public abstract class E2eTestBase {
+
+    private static final Logger LOG = LoggerFactory.getLogger(E2eTestBase.class);
+
+    // ------------------------------------------------------------------------------------------
+    // Flink Variables
+    // ------------------------------------------------------------------------------------------
+    // TODO change image tag to official flink image after 1.15 is released
+    private static final String FLINK_IMAGE_TAG = "tsreaper/flink-test:1.15-SNAPSHOT";
```
Comment (JingsongLi) [inline flink-table-store-e2e-tests/src/test/java/org/apache/flink/table/store/tests/E2eTestBase.java:60]: Can we read version from environment from pom `flink.version`?

Comment (JingsongLi) [inline flink-table-store-e2e-tests/src/test/java/org/apache/flink/table/store/tests/E2eTestBase.java:60]: Can we read version from environment?

**Review by JingsongLi — COMMENTED** [review]
Thanks @tsreaper for the contribution. Left some comments.

File: flink-table-store-e2e-tests/pom.xml (hunk 2 above)
Comment (tsreaper) [reply flink-table-store-e2e-tests/pom.xml:31]: This is actually unexpected. On my local machine the whole test runs for only 5 minutes. I'll look into this issue.

**Review by JingsongLi — APPROVED** [review]
Looks good to me!

**Pull request merged**